heft-plugins/heft-storybook-plugin/src/StorybookRunner.ts

```diff
@@ -29,6 +30,10 @@ export class StorybookRunner extends SubprocessRunnerBase<IStorybookRunnerConfig
 
     terminal.writeLine('Launching ' + this._configuration.resolvedStartupModulePath);
 
+    if (this._configuration.options) {
+      process.argv.push(...this._configuration.options);
```

Comment:
Hmm... I like the idea behind this, but I'm a bit worried about modifying the process.argv since that may be used by other plugins/functionality while running. Since it uses the subprocess runner, it's not really a concern in this version of Heft since it only modifies the subprocess's argv. However, the subprocess runner was removed in the in-development version of Heft and that means that storybook would run in-process, and any modifications to the argv could affect other plugins/functions that directly access the `process.argv`. This is also true when `heft --debug` is used, since the subprocess runner requires the module in-process there as well.

The fact that this module is blindly looking at the argv for this parameter (even though the command that is being run isn't applicable) is an indication that it's not designed very well to begin with. Ideally, we would require the module, then call some exported function to execute the main functionality.

It may be best to leave the path as-is for now.

Reply:
what about we do a teardown after requiring the startupModule?

```
// cache the original process.argv
const originalArgv = [...process.argv]

process.argv.push(...this._configuration. Options);
require(this._configuration.resolvedStartupModulePath);
terminal.writeVerboseLine('Completed synchronous portion of launching startupModulePath');

// teardown
process.argv = originalArgv
```

Reply:
Yes, good point. It's a synchronous require call, no other processes should be affected.